authorPatchCollector.py

```diff
@@ -0,0 +1,64 @@
+import sys
+import subprocess
+
+total_parameters = len(sys.argv)
+
+if(total_parameters != 3):
+    print(f'Usage: {str(sys.argv[0])} author_name output_root_directory')
+    print()
+    print("This is a simple helper script that allows collecting all patches from")
+    print("one author in a git repository. It will detect consequent ranges of")
+    print("patches and ask for each range for a subdirectory so you can distinguish")
+    print("the patch series'.")
+    sys.exit()
+
+author = str(sys.argv[1])
+output = str(sys.argv[2])
+end = ""
+begin = ""
+patch_count = 1
+range_count = 1
+
+get_hashes_cmd = f'git log --pretty=format:"%h" --author={author}'
+hashes = subprocess.check_output(get_hashes_cmd, shell=True)
+hashes = hashes.decode(encoding="utf-8")
+hashes = hashes.split("\n")
+
+for hash in hashes:
+    ancestor_hash_cmd = f'git log --pretty=format:"%h" --author={author} {hash}^^..{hash}^ --exit-code 1>/dev/null'
+    if (subprocess.Popen(ancestor_hash_cmd, shell=True)).wait():
+        if(end == ""):
+            end = hash
+        begin = hash
+        range_count = range_count + 1
+    else:
+        print("\n")
+        if(end == ""):
```

Comment:
I guess by now you know what to do :stuck_out_tongue_winking_eye:

Reply:
Yep, thank you :smile: